code/game/machinery/dye_generator.dm

```diff
@@ -94,6 +94,9 @@
 	if(!(M in view(1)))
 		..()
 		return
+	if(!Adjacent(user))
```

Comment:
won't fix the issue; you have to be adjacent to use this anyway; the problem is when people use it, open the menu, then move away and are no longer adjacent that the problem manifests.

Reply:
Ahhhh. I might keep the check for this anyway. will look at the menu stuff tomarrowish.